### ISSUE TITLE
Custom UI

### ISSUE THREAD

**Issue by c-mella** [chintan9/plyr-react#711]
Hi, 

I'm trying to build a custom UI around the audio player but finding that whenever i set a value using the useState hook it resets my player. Can you take a look and let me know if there's an easier way to do this? Either way I'd love if in a future version there was a way to just sub in the buttons, the same way we can set the positions and controls we want to see.

Thanks!

CODE BELOW

===========================================================================
```
/* eslint-disable new-cap */
import React, { useEffect, useRef, useState } from "react";
import PropTypes from "prop-types";
import plyr from 'plyr';
import "plyr-react/dist/plyr.css";
import Loadable from "@loadable/component";
import styles from "./PodcastPlayer.module.scss";

import initMoatTracking from "./initMoatTracking";
import extract from "src/shared/data/extract";
// import Tracker from "src/shared/util/tracker";

import toDashCase from "src/shared/util/toDashCase";

class PlyrComponent extends React.Component {

    componentDidMount() {
        this.player = new plyr(".js-plyr", this.props.options);
        this.player.source = this.props.file;
    }

    componentWillUnmount() {
        this.player.destroy();
    }

    render() {
        return (
            <audio className="js-plyr plyr" />
        );
    }

}

export default function PodcastPlayer({ content,
    bodyFont,
    subFont,
    advertiser,
    campaign,
    url,
    title,
    uid }) {

    if (!content) { return null }

    const file = extract(content, "audio_file.url");
    const contentId = extract(content, "audio_file.uid");

    const { description, style, publish_date, thumbnail, headline } = content || {};

    const { background_color, border_color, button_color, text_color } = style || {};

    const [dur, setDur] = useState();
    const [player, setPlayer] = useState();
    const [isPlaying, setPlaying] = useState(false);

    let muted = true;

    const options = {
        controls: [
          'rewind',
          'play',
          'fast-forward',
          'current-time',
          'progress',
          'duration',
          'mute',
          'volume',
          'settings',
          'fullscreen',
        ],
        autoplay: true,
        muted: false,
    }

    return (
        <div
            id={ contentId }
            style={{
                border: border_color && `2px solid ${ border_color }`,
                background: background_color,
                color: text_color
            }}
            className={ styles.audio } >

            <div
                className={ styles.container } >
                <img
                    src={ thumbnail.url }
                    className={ styles.thumb } />

                <div className={ styles.text }>
                    <div className={ styles.info }>
                        <svg className={ styles.calendar } fill="none" viewBox="0 0 11 11">
                            <path fillRule="evenodd" clipRule="evenodd" d="M9.286 2.452H2.211a.486.486 0 0 0-.486.487v6.685c0 .268.217.486.486.486h7.075a.486.486 0 0 0 .486-.486V2.939a.486.486 0 0 0-.486-.487Zm-7.075-.778c-.698 0-1.264.566-1.264 1.265v6.685c0 .698.566 1.264 1.264 1.264h7.075c.698 0 1.264-.566 1.264-1.264V2.939c0-.699-.566-1.265-1.264-1.265H2.211Z" fill="#000"/>
                            <rect x="2.799" y=".888" width=".751" height="2.273" rx=".376" fill="#000"/>
                            <rect x="7.938" y=".888" width=".751" height="2.273" rx=".376" fill="#000"/>
                            <path fill="#000" d="M1.45 4.052h8.701v.736H1.45z"/>
                            <rect x="2.272" y="5.545" width="7.058" height=".736" rx=".368" fill="#000"/>
                            <rect x="2.272" y="7.077" width="7.058" height=".736" rx=".368" fill="#000"/>
                            <rect x="2.272" y="8.61" width="7.058" height=".736" rx=".368" fill="#000"/>
                        </svg>
                        <p style={{ fontFamily: bodyFont }} className={ styles.pub } >{ publish_date }</p>
                        { dur > 0 &&
                            <svg className={ styles.clock } fill="none" viewBox="0 0 10 11">
                                <path fillRule="evenodd" clipRule="evenodd" d="M4.348 3.853a.39.39 0 0 1 .391.39v2.003h1.303a.39.39 0 1 1 0 .782H4.368a.41.41 0 0 1-.41-.41V4.242a.39.39 0 0 1 .39-.39Z" fill="#000"/>
                                <path fillRule="evenodd" clipRule="evenodd" d="M4.368 10.107a3.475 3.475 0 1 0 0-6.951 3.475 3.475 0 0 0 0 6.95Zm0 .781a4.257 4.257 0 1 0 0-8.514 4.257 4.257 0 0 0 0 8.514Z" fill="#000"/>
                                <rect x="3.228" y=".888" width="2.261" height=".784" rx=".392" fill="#000"/>
                                <rect x="4.75" y="1.024" width="1.846" height=".784" rx=".392" transform="rotate(90 4.75 1.024)" fill="#000"/>
                                <rect x="7.614" y="1.765" width="2.261" height=".784" rx=".392" transform="rotate(45 7.614 1.765)" fill="#000"/>
                                <rect x="8.595" y="2.938" width="1.846" height=".784" rx=".392" transform="rotate(135 8.595 2.938)" fill="#000"/>
                            </svg> }
                        {/* <p style={{ fontFamily: bodyFont }}
                            className={ styles.dur } >{ dur > 0 && toTime(dur) }</p> */}
                    </div>
                    <h3 style={{ fontFamily: subFont }}> { headline }</h3>
                    { description && <p style={{ fontFamily: bodyFont }}>{ description }</p> }

                    <div className={styles.controls}>

                        <button
                            className={ styles.back }
                            onClick={ () => { player.rewind(10) }}>
                            <svg fill="none" viewBox="0 0 22 22">
                                <path d="M.199 10.993a10.443 10.443 0 0 0 .824 4.087 10.477 10.477 0 0 0 2.25 3.337 10.471 10.471 0 0 0 7.421 3.076 10.427 10.427 0 0 0 4.088-.826 10.464 10.464 0 0 0 3.344-2.25 10.468 10.468 0 0 0 3.075-7.424 10.438 10.438 0 0 0-.825-4.085 10.461 10.461 0 0 0-2.25-3.34A10.46 10.46 0 0 0 10.701.493 10.466 10.466 0 0 0 3.28 3.568c-.267.266-.52.547-.757.84V1.512a.453.453 0 0 0-.906.002V6.02H6.12a.454.454 0 0 0 .321-.773.455.455 0 0 0-.319-.133H3.188a9.526 9.526 0 0 1 7.513-3.66 9.543 9.543 0 0 1 6.746 16.288 9.542 9.542 0 0 1-16.29-6.748v-.134a.478.478 0 0 0-.958-.01V10.994Z" fill="#000"/>
                                <path d="M9.792 13.508v.622H6.219v-.622h1.413V8.64l-1.314.837v-.711l1.638-1.044h.423v5.786h1.413ZM12.713 14.201c-.732 0-1.29-.276-1.674-.828-.378-.557-.567-1.373-.567-2.447s.19-1.887.567-2.44c.384-.557.942-.836 1.674-.836s1.287.279 1.665.837c.384.552.576 1.362.576 2.43 0 1.074-.192 1.89-.576 2.447-.378.558-.933.837-1.665.837Zm0-.62c.51 0 .885-.216 1.125-.649.246-.431.37-1.104.37-2.016 0-.905-.12-1.571-.36-1.997-.24-.426-.619-.64-1.135-.64-.51 0-.888.214-1.134.64-.24.426-.36 1.091-.36 1.997 0 .906.12 1.579.36 2.016.246.432.624.649 1.134.649Z" fill="#000"/>
                            </svg>
                        </button>

                        <button
                            className={ styles.forward }
                            // disabled={ dur < cT + 30 }
                            onClick={ () => {
                                player.forward(30);
                            }}>
                            <svg fill="none" viewBox="0 0 22 22">
                                <path d="M21.198 11.437a10.444 10.444 0 0 1-3.073 7.425 10.47 10.47 0 0 1-7.422 3.075 10.427 10.427 0 0 1-4.088-.826 10.466 10.466 0 0 1-3.344-2.25 10.466 10.466 0 0 1-3.075-7.424 10.438 10.438 0 0 1 .825-4.084 10.46 10.46 0 0 1 2.25-3.34A10.46 10.46 0 0 1 10.696.937a10.466 10.466 0 0 1 7.422 3.075c.267.267.52.547.757.841V1.956a.453.453 0 0 1 .906.002v4.506h-4.505a.455.455 0 0 1-.321-.774.455.455 0 0 1 .319-.132h2.936a9.527 9.527 0 0 0-7.514-3.66 9.543 9.543 0 0 0-9.54 9.54 9.543 9.543 0 0 0 19.085 0v-.135a.478.478 0 0 1 .957-.009V11.437Z" fill="#000"/>
                                <path d="M9.107 11.28c.378.108.666.294.864.558.198.264.297.597.297.999 0 .558-.189.999-.567 1.323-.378.318-.89.477-1.539.477-.414 0-.813-.07-1.197-.207a2.73 2.73 0 0 1-.963-.585l.288-.567c.594.492 1.212.738 1.854.738.456 0 .801-.102 1.035-.306.24-.204.36-.504.36-.9 0-.786-.492-1.18-1.476-1.18h-.738v-.62h.594c.474 0 .837-.105 1.09-.315.251-.21.377-.51.377-.9 0-.348-.105-.615-.315-.801-.204-.186-.495-.28-.873-.28-.612 0-1.218.247-1.818.739l-.279-.567a2.72 2.72 0 0 1 .945-.576 3.284 3.284 0 0 1 1.18-.216c.575 0 1.034.147 1.376.44.348.295.522.69.522 1.189 0 .366-.09.687-.27.963-.18.276-.429.474-.747.594ZM13.172 14.646c-.732 0-1.29-.276-1.674-.828-.378-.558-.567-1.374-.567-2.448 0-1.074.189-1.887.567-2.44.384-.557.942-.836 1.674-.836s1.287.279 1.665.837c.384.552.576 1.362.576 2.43 0 1.074-.192 1.89-.576 2.448-.378.558-.933.837-1.665.837Zm0-.621c.51 0 .885-.216 1.125-.648.246-.432.369-1.104.369-2.016 0-.906-.12-1.572-.36-1.998-.24-.426-.618-.64-1.134-.64-.51 0-.888.214-1.134.64-.24.426-.36 1.092-.36 1.998 0 .906.12 1.578.36 2.016.246.432.624.648 1.134.648Z" fill="#000"/>
                            </svg>
                        </button>

                        <button
                            className={ styles.play }
                            onClick={ () => {
                                PlyrComponent.togglePlay();
                                setPlaying(!isPlaying);
                            }}>

                            { isPlaying
                                ? <svg fill="none" viewBox="0 0 35 34">
                                    <circle cx="17.198" cy="17.057" r="16.817" fill={ button_color }/>
                                    <path fill={ background_color } d="M12.604 11.557h3.38v10.999h-3.38zM18.414 11.557h3.38v10.999h-3.38z"/>
                                </svg>
                                : <svg fill="none" viewBox="0 0 35 35">
                                    <circle cx="17.198" cy="17.806" r="16.817" fill={ button_color }/>
                                    <path d="m22.849 17.806-8.476 4.893v-9.786l8.476 4.893Z" fill={ background_color }/>
                                </svg>
                            }
                        </button>

                        {/* <span
                            style={{ fontFamily: bodyFont }}
                            className={ styles.current }>
                            { player && toTime(player.currentTime) } </span> */}

                        <input
                            className={ styles.range }
                            type="range"
                            min="0" max={ dur }
                            // value={ cT }
                            onChange={ (e) => {
                                player.currentTime = e.target.value;
                                player.muted = false;
                                player.play();
                            }}
                            step=".25"
                        />

                        {/* <span
                            style={{ fontFamily: bodyFont }}
                            className={ styles.duration }> { player && toTime(dur) } </span> */}

                        <button
                            className={ styles.volume }
                            onClick={ () => { player.muted = !muted } }>
                            { muted
                                ? <svg fill="none" viewBox="0 0 12 17">
                                    <path d="M0 6.415c.205-.64.664-.866 1.322-.847.852.025 1.705.011 2.558 0a.624.624 0 0 0 .36-.127C6.064 3.961 7.885 2.476 9.703.987c.54-.44 1.197-.409 1.574.112.135.184.201.443.215.676.031.53.01 1.062.01 1.592v12.101c0 .652-.347 1.08-.934 1.154-.326.04-.597-.083-.843-.284a3947.75 3947.75 0 0 0-4.147-3.387c-.45-.367-.894-.74-1.35-1.097a.605.605 0 0 0-.339-.119c-.838-.01-1.676-.03-2.513.003-.658.026-1.136-.173-1.37-.809L0 6.415Z" fill={ button_color }/>
                                </svg>
                                : <svg fill="none" viewBox="0 0 23 19">
                                    <path fill={ button_color } d="M0 6.941c.205-.64.664-.866 1.322-.847.852.025 1.705.012 2.558 0a.624.624 0 0 0 .36-.126 1509.22 1509.22 0 0 0 5.463-4.455c.54-.44 1.197-.409 1.574.112.135.184.201.443.215.676.031.53.01 1.062.01 1.592v12.102c0 .65-.347 1.079-.934 1.153-.326.04-.597-.082-.843-.284-1.382-1.13-2.764-2.259-4.147-3.387-.45-.367-.894-.74-1.35-1.097a.606.606 0 0 0-.339-.119c-.838-.01-1.676-.03-2.513.003-.658.026-1.136-.173-1.37-.808L0 6.94ZM23 8.843c-.054 3.512-1.07 6.33-3.108 8.78-.416.5-1.035.585-1.507.215-.593-.462-.71-1.054-.283-1.597.535-.684 1.037-1.387 1.426-2.164a10.724 10.724 0 0 0 1.074-3.47c.365-2.861-.31-5.47-1.992-7.825-.206-.288-.441-.557-.668-.832-.457-.552-.379-1.153.219-1.641.47-.386 1.095-.303 1.53.191 1.624 1.857 2.67 3.988 3.09 6.405.128.753.173 1.52.219 1.938Z"/>
                                    <path fill={ button_color } d="M17.911 9.16c.02 1.814-.53 3.45-1.612 4.914-.347.469-.962.584-1.433.28-.58-.377-.76-.996-.459-1.529.3-.532.665-1.045.868-1.613.709-1.976.432-3.84-.792-5.563-.152-.214-.311-.444-.381-.689-.138-.488.193-1.03.714-1.268.437-.2.948-.076 1.268.32 1.002 1.24 1.615 2.641 1.79 4.222.01.096.033.19.035.285.006.214.002.428.002.642Z"/>
                                </svg> }
                        </button>
                    </div>

                    <PlyrComponent
                        options={ options }
                        file={ file }
                    />

                </div>
            </div>
        </div>
    );
}

PodcastPlayer.propTypes = {
    audio: PropTypes.object,
    fontFamily: PropTypes.string,
    options: PropTypes.object,
    advertiser: PropTypes.string,
    campaign: PropTypes.string,
    url: PropTypes.string,
    title: PropTypes.string,
    uid: PropTypes.string
};

```

**Comment by amirHossein-Ebrahimi** [comment]
Sorry for the delay, but we now pre-lease a new hook for our proposal method that helps you to be as low level as the child component (Plyr video), it helps create more custom components easily with `usePlyr` hook.

Feel free to checkout [example directory](https://github.com/chintan9/plyr-react/tree/dev/example).

**Comment by c-mella** [comment]
@amirHossein-Ebrahimi  Sorry to bother you again but when will this be officially released? In the meantime I can just use the pre-lease you sent me but I just wanted to see if you have a timeline

**Comment by amirHossein-Ebrahimi** [comment]
Hello, @c-mella. It is now available in npm with the alpha tag [plyr-react@4.0.0-alpha] .1] (https://www.npmjs.com/package/plyr-react/v/4.0.0-alpha.1)

The `usePlyr` method is now functioning as intended, but we need further feedback to ensure that this API is adequate and will not require new breaking changes (we're all weary of breaking changes).

However, I believe you should consider the new version to be a release candidate. The only thing I intend to alter is to make `usePlyr` more generic and controlled using a dependency array, but this will take more time to test.
It will provide you with a lot of power as a developer, but it requires you to understand exactly what happens beneath the hood of `plyr` and `react-aptor`.

**Comment by c-mella** [comment]
@amirHossein-Ebrahimi Thanks again for all your help, this library is awesome
